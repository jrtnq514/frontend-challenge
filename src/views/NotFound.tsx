import { Box, Center, Heading } from "@chakra-ui/react";
import React from "react";

const NotFound: React.FC = () => {
  return (
    <Box p={16}>
      <Center>
        <Heading as="h1">Page Not Found</Heading>
      </Center>
    </Box>
  );
};

export default NotFound;
