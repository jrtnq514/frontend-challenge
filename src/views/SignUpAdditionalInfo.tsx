import React, { useEffect } from "react";
import { Link, useLoaderData, useNavigate } from "react-router-dom";
import {
  Box,
  Heading,
  VStack,
  HStack,
  Button,
  Text,
  FormControl,
  FormLabel,
  FormErrorMessage,
  Select,
  Checkbox,
} from "@chakra-ui/react";
import { useSignUpState } from "src/contexts/signUpStateContext";
import { useForm, SubmitHandler } from "react-hook-form";

export interface SignUpAdditionalInfoInputs {
  favoriteColor: string;
  agreeTermsAndConditions: boolean;
}

const SignUpAdditionalInfo: React.FC = () => {
  const navigate = useNavigate();
  const { signUpState, setSignUpState } = useSignUpState();
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors },
  } = useForm<SignUpAdditionalInfoInputs>({ defaultValues: { agreeTermsAndConditions: false }});
  const colors = useLoaderData() as string[];

  useEffect(() => {
    if (!signUpState.firstName || !signUpState.email || !signUpState.password) {
      navigate("/");
    }

    if (signUpState.favoriteColor) {
      setValue("favoriteColor", signUpState.favoriteColor);
    }
  }, []);

  const onSubmit: SubmitHandler<SignUpAdditionalInfoInputs> = (data) => {
    setSignUpState({
      ...signUpState,
      favoriteColor: data.favoriteColor,
      agreeTermsAndConditions: data.agreeTermsAndConditions
    });
    navigate("/confirmation");
  };

  return (
    <Box>
      <VStack spacing={6}>
        <Heading as="h1">Sign Up</Heading>
        <Text>Lets get started with some basic information.</Text>
        <Box as="form" width="100%" onSubmit={handleSubmit(onSubmit)}>
          <VStack spacing={4}>
            <FormControl isInvalid={!!errors.favoriteColor}>
              <FormLabel>Choose your favorite color</FormLabel>
              <Select
                placeholder="Choose a color"
                {...register("favoriteColor", { required: true })}
              >
                { colors.map((color) => <option value={color} key={color}>{color}</option>) }
              </Select>
              {errors.favoriteColor && errors.favoriteColor.type === "required" && (
                <FormErrorMessage>Favorite color is required.</FormErrorMessage>
              )}
            </FormControl>
            <FormControl isInvalid={!!errors.agreeTermsAndConditions}>
              <HStack>
                <Checkbox {...register("agreeTermsAndConditions", { required: true })}>
                  I Agree to <Link to="#">Terms and Conditions</Link>
                </Checkbox>
              </HStack>
              <FormErrorMessage>You must agree to the terms and conditions.</FormErrorMessage>
            </FormControl>
            <HStack justifyContent="space-between" width="100%">
              <Button type="button" variant="outline" colorScheme="green" onClick={(event) => navigate('/')}>Back</Button>
              <Button type="submit" variant="solid" colorScheme="green">
                Next
              </Button>
            </HStack>
          </VStack>
        </Box>
      </VStack>
    </Box>
  );
};

export default SignUpAdditionalInfo;
