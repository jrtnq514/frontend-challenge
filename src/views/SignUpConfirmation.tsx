import React, { FormEvent, useEffect, useState } from "react";
import {
  Box,
  Spinner,
  VStack,
  Heading,
  Text,
  List,
  ListItem,
  HStack,
  Button,
} from "@chakra-ui/react";
import { useNavigate, useNavigation } from "react-router-dom";
import { useSignUpState } from "src/contexts/signUpStateContext";
import request from "src/utils/requestClient";

const SignUpConfirmation: React.FC = () => {
  const navigate = useNavigate();
  const { state: navState } = useNavigation();
  const { signUpState } = useSignUpState();
  const [showPassword, setShowPassword] = useState(false);
  const [isSubmitting, setIsSubmitting] = useState(false);

  useEffect(() => {
    if (!signUpState.firstName || !signUpState.email || !signUpState.password) {
      navigate("/");
    }

    if (!signUpState.favoriteColor || !signUpState.agreeTermsAndConditions) {
      navigate("/more-info");
    }
  }, []);

  const submitSignUp = async (event: FormEvent) => {
    event.preventDefault();
    setIsSubmitting(true);
    try {
      await request.post("submit", {
        json: {
          name: signUpState.firstName,
          email: signUpState.email,
          password: signUpState.password,
          color: signUpState.favoriteColor,
          terms: signUpState.agreeTermsAndConditions,
        },
      });
      navigate("/success");
    } catch (error) {
      console.log(error);
      navigate("/error");
    } finally {
      setIsSubmitting(false);
    }
  };

  return (
    <Box>
      {navState === "loading" ? (
        <Spinner size="xl" />
      ) : (
        <Box
          as="form"
          data-testid="confirmation-form"
          width="100%"
          onSubmit={submitSignUp}
        >
          <VStack spacing={6}>
            <Heading as="h1">Confirmation</Heading>
            <Text>Make sure everything looks ok.</Text>
            <List fontSize="xl" spacing={4}>
              <ListItem>
                <HStack wrap="wrap">
                  <Box fontWeight="light">First Name:</Box>
                  <Box data-testid="first-name-value">
                    {signUpState.firstName}
                  </Box>
                </HStack>
              </ListItem>
              <ListItem>
                <HStack wrap="wrap">
                  <Box fontWeight="light">Email:</Box>
                  <Box data-testid="email-value">{signUpState.email}</Box>
                </HStack>
              </ListItem>
              <ListItem>
                <HStack wrap="wrap">
                  <Box fontWeight="light">Password:</Box>
                  <Box data-testid="password-value">
                    {showPassword
                      ? signUpState.password
                      : signUpState.password?.split("").map((x) => `â€¢`)}
                  </Box>
                  <Button
                    data-testid="show-password-btn"
                    size="sm"
                    variant="outline"
                    onClick={(event) => setShowPassword(!showPassword)}
                  >
                    {showPassword ? "Hide" : "Show"}
                  </Button>
                </HStack>
              </ListItem>
              <ListItem>
                <HStack wrap="wrap">
                  <Box fontWeight="light">Favorite Color:</Box>
                  <Box data-testid="color-value">
                    {signUpState.favoriteColor}
                  </Box>
                </HStack>
              </ListItem>
              <ListItem>
                <HStack wrap="wrap">
                  <Box fontWeight="light">Accepted Terms and Conditions: </Box>
                  <Box data-testid="terms-value">
                    {signUpState.agreeTermsAndConditions
                      ? "Accepted"
                      : "Declined"}
                  </Box>
                </HStack>
              </ListItem>
            </List>
            <HStack justifyContent="space-between" width="100%">
              <Button
                type="button"
                variant="outline"
                colorScheme="green"
                onClick={(event) => navigate("/more-info")}
              >
                Back
              </Button>
              <Button
                data-testid="submit-btn"
                type="submit"
                variant="solid"
                colorScheme="green"
                isDisabled={isSubmitting}
              >
                {isSubmitting ? (
                  <Spinner data-testid="submit-btn-spinner" />
                ) : (
                  "Submit"
                )}
              </Button>
            </HStack>
          </VStack>
        </Box>
      )}
    </Box>
  );
};

export default SignUpConfirmation;
