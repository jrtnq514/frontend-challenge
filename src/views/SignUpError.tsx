import React from 'react';

const SignUpError: React.FC = () => {

  return (
    <></>
  );
};

export default SignUpError;