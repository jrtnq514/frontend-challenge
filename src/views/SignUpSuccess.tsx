import React from 'react';

const SignUpSuccess: React.FC = () => {

  return (
    <></>
  );
};

export default SignUpSuccess;