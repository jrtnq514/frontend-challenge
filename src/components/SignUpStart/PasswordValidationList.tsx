import React from "react";
import {
  Card,
  CardHeader,
  CardBody,
  Text,
  List,
  ListItem,
  ListIcon,
} from "@chakra-ui/react";
import { CheckIcon, CloseIcon } from "@chakra-ui/icons";
import { FieldErrorsImpl } from "react-hook-form";
import { SignUpStartInputs } from "src/views/SignUpStart";

interface PasswordValidationListItemProps {
  isError: boolean;
  children: React.ReactNode;
}

const PasswordValidationListItem: React.FC<PasswordValidationListItemProps> = ({
  isError,
  children,
}) => {
  return (
    <ListItem color={isError ? "red.500" : "green.700"} display="flex">
      {isError ? <ListIcon as={CloseIcon} /> : <ListIcon as={CheckIcon} />}
      {children}
    </ListItem>
  );
};

interface PasswordValidationListProps {
  errors: FieldErrorsImpl<SignUpStartInputs>;
}

const PasswordValidationList: React.FC<PasswordValidationListProps> = ({
  errors,
}) => {
  return (
    <Card width="100%">
      <CardHeader>
        <Text fontWeight={600} as="label">Passwords must:</Text>
      </CardHeader>
      <CardBody pt={0}>
        <List spacing={0.5}>
          <PasswordValidationListItem
            isError={
              !!(
                errors.password?.types?.minLength ||
                errors.password?.types?.required
              )
            }
          >
            Be at least 8 characters long
          </PasswordValidationListItem>
          <PasswordValidationListItem
            isError={!!errors.password?.types?.passwordUppercase}
          >
            Contain a uppercase letter
          </PasswordValidationListItem>
          <PasswordValidationListItem
            isError={!!errors.password?.types?.passwordLowercase}
          >
            Contain a lowercase letter
          </PasswordValidationListItem>
          <PasswordValidationListItem
            isError={!!errors.password?.types?.passwordNumber}
          >
            Contain a number
          </PasswordValidationListItem>
          <PasswordValidationListItem
            isError={!!errors.password?.types?.passwordSpecial}
          >
            Contain a special character
          </PasswordValidationListItem>
        </List>
      </CardBody>
    </Card>
  );
};

export default PasswordValidationList;
