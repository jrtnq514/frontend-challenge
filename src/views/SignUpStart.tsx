import {
  Box,
  Heading,
  VStack,
  Text,
  FormLabel,
  FormControl,
  Input,
  HStack,
  Button,
  FormErrorMessage,
  InputRightElement,
  InputGroup,
  Popover,
  PopoverArrow,
  PopoverContent,
  PopoverCloseButton,
  PopoverHeader,
  PopoverBody,
  IconButton,
  useDisclosure,
  PopoverAnchor,
  List,
  ListItem,
  Spinner,
} from "@chakra-ui/react";
import { InfoIcon } from "@chakra-ui/icons";
import React, { useEffect, useState } from "react";
import { useNavigate, useNavigation } from "react-router-dom";
import { useSignUpState } from "src/contexts/signUpStateContext";
import { useForm, SubmitHandler } from "react-hook-form";
import PasswordValidationList from "src/components/SignUpStart/PasswordValidationList";

export interface SignUpStartInputs {
  firstName: string;
  email: string;
  password: string;
}

const SignUpStart: React.FC = () => {
  const navigate = useNavigate();
  const { state: navState } = useNavigation();
  const [showPassword, setShowPassword] = useState(false);
  const { isOpen, onToggle, onClose } = useDisclosure();
  const { signUpState, setSignUpState } = useSignUpState();
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors },
  } = useForm<SignUpStartInputs>({ criteriaMode: "all" });

  const passwordUppercase = (input: string) => /[A-Z]/.test(input);
  const passwordLowercase = (input: string) => /[a-z]/.test(input);
  const passwordNumber = (input: string) => /[0-9]/.test(input);
  const passwordSpecial = (input: string) =>
    /[~`! @#$%^&*()_\-+={[}\]|\:;"'<,>.?/]/.test(input);

  useEffect(() => {
    if (signUpState.firstName) {
      setValue("firstName", signUpState.firstName);
    }

    if (signUpState.email) {
      setValue("email", signUpState.email);
    }

    setSignUpState({
      ...signUpState,
      firstName: "",
      email: "",
      password: "",
    });
  }, []);

  useEffect(() => {
    if (
      signUpState.firstName &&
      signUpState.email &&
      signUpState.password
    ) {
      navigate("/more-info");
    }
  }, [signUpState]);

  const onSubmit: SubmitHandler<SignUpStartInputs> = (data) => {
    setSignUpState({
      ...signUpState,
      firstName: data.firstName,
      email: data.email,
      password: data.password,
    });
  };

  return (
    <Box>
      {navState === "loading" ? (
        <Spinner size="xl" />
      ) : (
        <VStack spacing={6}>
          <Heading as="h1">Sign Up</Heading>
          <Text>Lets get started with some basic information.</Text>
          <Box
            as="form"
            width="100%"
            data-id="start-form"
            onSubmit={handleSubmit(onSubmit)}
          >
            <VStack spacing={4}>
              <FormControl isInvalid={!!errors.firstName}>
                <FormLabel>First Name</FormLabel>
                <Input
                  data-id="first-name-input"
                  type="text"
                  {...register("firstName", { required: true })}
                />
                {errors.firstName && errors.firstName.types?.required && (
                  <FormErrorMessage>First Name is required.</FormErrorMessage>
                )}
              </FormControl>
              <FormControl isInvalid={!!errors.email}>
                <FormLabel>Email</FormLabel>
                <Input
                  data-id="email-input"
                  type="email"
                  {...register("email", { required: true })}
                />
                {errors.email && errors.email.types?.required && (
                  <FormErrorMessage>Email is required.</FormErrorMessage>
                )}
              </FormControl>
              <FormControl isInvalid={!!errors.password}>
                <FormLabel>
                  Password{" "}
                  <Popover isOpen={isOpen} onClose={onClose} placement="auto">
                    <PopoverAnchor>
                      <IconButton
                        onClick={onToggle}
                        aria-label="Show password requirements"
                        height="auto"
                        minW="initial"
                        icon={<InfoIcon />}
                      />
                    </PopoverAnchor>
                    <PopoverContent>
                      <PopoverArrow />
                      <PopoverCloseButton />
                      <PopoverHeader>Passwords must:</PopoverHeader>
                      <PopoverBody>
                        <List>
                          <ListItem>Be at least 8 characters long</ListItem>
                          <ListItem>Contain a uppercase letter</ListItem>
                          <ListItem>Contain a lowercase letter</ListItem>
                          <ListItem>Contain a number</ListItem>
                          <ListItem>Contain a special character</ListItem>
                        </List>
                      </PopoverBody>
                    </PopoverContent>
                  </Popover>
                </FormLabel>
                <InputGroup>
                  <Input
                    data-id="password-input"
                    type={showPassword ? "text" : "password"}
                    pr={12}
                    {...register("password", {
                      required: true,
                      minLength: 8,
                      validate: {
                        passwordUppercase,
                        passwordLowercase,
                        passwordNumber,
                        passwordSpecial,
                      },
                    })}
                  />
                  <InputRightElement width={12}>
                    <Button
                      size="sm"
                      variant="ghost"
                      onClick={(event) => setShowPassword(!showPassword)}
                    >
                      {showPassword ? "Hide" : "Show"}
                    </Button>
                  </InputRightElement>
                </InputGroup>
                <FormErrorMessage>
                  <PasswordValidationList errors={errors} />
                </FormErrorMessage>
              </FormControl>
              <HStack width="100%" justifyContent="end">
                <Button type="submit" variant="solid" colorScheme="green">
                  Next
                </Button>
              </HStack>
            </VStack>
          </Box>
        </VStack>
      )}
    </Box>
  );
};

export default SignUpStart;
