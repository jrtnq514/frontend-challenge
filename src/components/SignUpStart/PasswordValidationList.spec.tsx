import PasswordValidationList from "./PasswordValidationList";
import { render, screen, waitFor } from "@testing-library/react";
import { FieldErrorsImpl } from "react-hook-form";
import { SignUpStartInputs } from "src/views/SignUpStart";

describe("PasswordValidationList", () => {
  it("should render all validation items as errored", async () => {
    const errors: FieldErrorsImpl<SignUpStartInputs> = {
      password: {
        message: "",
        type: "required",
        types: {
          required: true,
          passwordLowercase: true,
          passwordNumber: true,
          passwordSpecial: true,
          passwordUppercase: true,
        },
      },
    };

    const { container } = render(<PasswordValidationList errors={errors} />);

    await waitFor(() =>
      screen.findAllByTestId("password-validation-list-item-error")
    );

    const listErrors = await screen.findAllByTestId(
      "password-validation-list-item-error"
    );
    expect(listErrors.length).toEqual(5);
    expect(container).toMatchSnapshot();
  });

  it("should render all validation items as valid", async () => {
    const errors: FieldErrorsImpl<SignUpStartInputs> = {
      password: {
        message: "",
        type: "required",
        types: {},
      },
    };

    const { container } = render(<PasswordValidationList errors={errors} />);

    await waitFor(() =>
      screen.findAllByTestId("password-validation-list-item-valid")
    );

    const listValids = await screen.findAllByTestId(
      "password-validation-list-item-valid"
    );
    expect(listValids.length).toEqual(5);
    expect(container).toMatchSnapshot();
  });
});
