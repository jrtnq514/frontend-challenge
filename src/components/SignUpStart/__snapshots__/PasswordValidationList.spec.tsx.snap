// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordValidationList should render all validation items as errored 1`] = `
<div>
  <div
    class="chakra-card css-stnqs5"
  >
    <div
      class="chakra-card__header css-0"
    >
      <label
        class="chakra-text css-35ezg3"
      >
        Passwords must:
      </label>
    </div>
    <div
      class="chakra-card__body css-1161qt5"
    >
      <ul
        class="css-1f8kndt"
        role="list"
      >
        <li
          class="css-1o9fset"
          data-testid="password-validation-list-item-error"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
          Be at least 8 characters long
        </li>
        <li
          class="css-1o9fset"
          data-testid="password-validation-list-item-error"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
          Contain a uppercase letter
        </li>
        <li
          class="css-1o9fset"
          data-testid="password-validation-list-item-error"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
          Contain a lowercase letter
        </li>
        <li
          class="css-1o9fset"
          data-testid="password-validation-list-item-error"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
          Contain a number
        </li>
        <li
          class="css-1o9fset"
          data-testid="password-validation-list-item-error"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
          Contain a special character
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`PasswordValidationList should render all validation items as valid 1`] = `
<div>
  <div
    class="chakra-card css-stnqs5"
  >
    <div
      class="chakra-card__header css-0"
    >
      <label
        class="chakra-text css-35ezg3"
      >
        Passwords must:
      </label>
    </div>
    <div
      class="chakra-card__body css-1161qt5"
    >
      <ul
        class="css-1f8kndt"
        role="list"
      >
        <li
          class="css-1b3cqhx"
          data-testid="password-validation-list-item-valid"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
          Be at least 8 characters long
        </li>
        <li
          class="css-1b3cqhx"
          data-testid="password-validation-list-item-valid"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
          Contain a uppercase letter
        </li>
        <li
          class="css-1b3cqhx"
          data-testid="password-validation-list-item-valid"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
          Contain a lowercase letter
        </li>
        <li
          class="css-1b3cqhx"
          data-testid="password-validation-list-item-valid"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
          Contain a number
        </li>
        <li
          class="css-1b3cqhx"
          data-testid="password-validation-list-item-valid"
        >
          <svg
            class="chakra-icon chakra-icon css-ikjmp3"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
          Contain a special character
        </li>
      </ul>
    </div>
  </div>
</div>
`;
